go/types/blob.go

```diff
@@ -295,6 +295,7 @@ func readBlob(r io.Reader, vrw ValueReadWriter) Blob {
 	mtChan := make(chan chan metaTuple, runtime.NumCPU())
 
 	makeChunk := func() {
+		rv.Reset()
```

Comment:
It's a long time ago, but I my memory is that we did have that concern, but I think it wasn't well understood. Logically,  this change *reduces* the degree to which changes can cause cascade chunks because a boundary is only dependent on the N previous bytes *since the last boundary. One important change here is that if a change occurs near the end of the boundary, but doesn't move it, we dont *need* to enter the next chunk to see if it changed because it can't have.

The thing to be worried about with this change is the opposite -- that it reduces entropy. The first bytes of a chunk are suspectible to chunking in the same way if a small number of bytes are repeated because the "previous" bytes are effectively all zeros. That's partially why the change to struct encoding is included in this patch (IOW, to reduce the risk that a struct with field name bytes > 64) happens to chunk and causes a leaf level of only 1 element nodes.

In either case, I instrumented attic to report the rate of chunk cascades before & after this change. Since attic isn't deterministic, the numbers aren't stable, but using 1000 photos going through the pipeline, before & after, the cascade rate was ~.8%.